File: missingNumber.py

# leetcode.com 268. Missing Number


class Solution:
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()

        if nums[-1] != len(nums):
            return len(nums)

        if nums[0] != 0:
            return 0

        for i in range(1, len(nums)):
            expected_num = nums[i - 1] + 1
            if nums[i] != expected_num:
                return expected_num

File: leetcode/java/src/main/java/com/vkeonline/leetcode/p200/MissingNumber.java
package com.vkeonline.leetcode.p200;

import java.util.Arrays;

/**
 * Leetcode [E]: 268. Missing Number
 * @author csgear
 */
public class MissingNumber {
    public int missingNumber(int[] nums) {
        int sum = Arrays.stream(nums).sum() ;
        int n = nums.length ;
        return n * (n + 1) / 2 - sum ;
    }
}
